### ISSUE TITLE
`consistent-destructuring` should report error when destructure same object twice

### ISSUE THREAD

**Issue by zanminkian** [sindresorhus/eslint-plugin-unicorn#2364]
1️⃣ Explain here what's wrong

Destructuring an object implies that this object will not be used later.

```js
const {foo, bar} = baz;
// from then on, code below will not use `baz` any more
```

Therefore, `consistent-destructuring` will report error on the code

```js
const {foo} = baz;

// ... other code

console.log(baz.bar); // will report error this line. it's expected
```

However, destructuring same object twice will not cause error, which is unexpected.

```js
const {foo} = baz;

// ... other code

const {bar} = baz; // will not report error this line. it's unexpected
console.log(bar);
```

2️⃣ Specify which rule is buggy here and in the title

consistent-destructuring

3️⃣ Add some examples where the issue appears

```js
const {foo} = baz;

// ... other code

const {bar} = baz;
console.log(bar);
```

eslint should report error on the code above when enabling `consistent-destructuring`.

**Comment by fregante** [comment]
👎 

I think multiple destructuring are fine for the same reason why manual hoisting isn't preferred anymore: just create variables as needed, you don't need to create them all at the beginning.

**Comment by sindresorhus** [comment]
I agree with @fregante.
